### ISSUE TITLE
[BUG] startup failure caused by dependencies, requires httpx

### ISSUE THREAD

**Issue by Bidaya0** [schemathesis/schemathesis#1614]
**Checklist**

- [x] I checked the [FAQ section](https://schemathesis.readthedocs.io/en/stable/faq.html#frequently-asked-questions) of the documentation
- [x] I looked for similar issues in the [issue tracker](https://github.com/schemathesis/schemathesis/issues)

**Describe the bug**
Because the upstream starlette updated the dependencies,the following error occurs
ModuleNotFoundError: No module named 'httpx'

**To Reproduce**

https://github.com/HXSecurity/DongTai/actions/runs/3132544194/jobs/5084991907
  File "/home/runner/.local/bin/schemathesis", line 5, in <module>
    from schemathesis.cli import schemathesis
  File "/home/runner/.local/lib/python3.8/site-packages/schemathesis/__init__.py", line 7, in <module>
    from . import auth, fixups, graphql, hooks, runner, serializers, targets
  File "/home/runner/.local/lib/python3.8/site-packages/schemathesis/graphql.py", line 3, in <module>
    from .specs.graphql import nodes
  File "/home/runner/.local/lib/python3.8/site-packages/schemathesis/specs/graphql/__init__.py", line 1, in <module>
    from .loaders import from_asgi, from_dict, from_file, from_path, from_url, from_wsgi
  File "/home/runner/.local/lib/python3.8/site-packages/schemathesis/specs/graphql/loaders.py", line 9, in <module>
    from starlette.testclient import TestClient as ASGIClient
  File "/home/runner/.local/lib/python3.8/site-packages/starlette/testclient.py", line 15, in <module>
    import httpx
ModuleNotFoundError: No module named 'httpx'

**Expected behavior**
It seems that dependencies that are still in version 0.x.y should not be used openly, which means their api is still changing frequently.

**Environment (please complete the following information):**
 - OS:Linux
 - Python version: 3.8
 - Schemathesis version: 3.17.2

**Additional context**
looks related to this
https://github.com/encode/starlette/pull/1376

It looks like this issue is not fully resolved.
https://github.com/schemathesis/schemathesis/issues/1417

**Comment by t-huyeng** [comment]
I can confirm this issue.
* using GitHub Actions

**Comment by TheMatrix97** [comment]
facing the same issue, no matter what version of schemathesis / python

**Comment by Kludex** [comment]
`pip install httpx` is a temporary fix 👀

**Comment by mdavis-xyz** [comment]
The [starlette readme](https://github.com/encode/starlette) says:

> Starlette only requires anyio, and the following are optional:
> *     [httpx](https://www.python-httpx.org/) - Required if you want to use the TestClient.

It looks like Schemathesis wants to use the TestClient, therefore httpx should be added as a non-temporary requirement of schemathesis. I'll write a PR for that.

**Comment by mdavis-xyz** [comment]
Ah actually it looks like schemathesis uses poetry, which I am not familiar with. So I'm not confident I know where to add the new dependency.

Can someone else add it? Does it go in pyproject.toml or poetry.lock?

**Comment by Kludex** [comment]
`poetry add httpx>=0.22.0`
`poetry lock --no-update`

**Comment by mdavis-xyz** [comment]
Ok I'll try that.

Why did you say `httpx>=0.22.0` and not just `httpx`?

**Comment by Kludex** [comment]
Because it's the minimum version that Starlette supports. 🙏

**Comment by CommanderStorm** [comment]
> `pip install httpx` is a temporary fix eyes

This is awkward: How did you do this?
I [tried to add this dependency via pip this way](https://github.com/TUM-Dev/NavigaTUM/blob/230dd80c5e6a5bdd03887eecc5ab657175dcfef8/.github/workflows/server.yml#L95), but this does not work, as (assuming this is the culprit) the schemathesis action does install python dependencies itself.

**Comment by Kludex** [comment]
idk, different virtual env? :man_shrugging:

**Comment by devicehzy** [comment]
*  solution by poetry  
```
[tool.poetry.dependencies]
starlette = "0.17.1"
werkzeug = "2.0.2"
```

**Comment by Stranger6667** [comment]
The issue is fixed in `3.17.3`. Let me know if the fix doesn't work for you

**Comment by Stranger6667** [comment]
And sorry for the delay!
